keps/sig-node/20190717-seccomp-ga.md

```diff
@@ -193,31 +218,65 @@ for new profile sources to be added in the future (e.g. Kubernetes predefined pr
 profiles). The seccomp options struct leaves room for future extensions, such as defining the
 behavior when a profile cannot be set.
 
-<<[UNRESOLVED]>>
-What to do with the localhost profile type, given that we want to deprecate it? Alternative for
-consideration:
+##### LocalhostProfile
+
+This KEP proposes we GA LocalhostProfile as the only source of user-defined profiles at this point. 
+User-defined profiles are essential for users to realize the full benefits out of seccomp, allowing 
+them to decrease their attack surface based on their own workloads. 
+
+###### Updating seccomp profiles
+Seccomp profiles are applied at container creation time. Therefore, updating the file contents of a 
+profile on disk after that point will not cause any changes to the running containers that are using 
+it, but will apply the updated profile on container restart. Note that amending such files is not 
+recommended and may cause containers to fail on next restart, in the case of the new profile being more 
+restrictive, invalid or the file no longer being present on disk.
+
+Currently, users have no way to tell whether their physical profiles have been deleted or modified.
+This KEP proposes no changes to the existing functionality.
+
+The recommended approach for rolling out changes to seccomp profiles is to always create _new profiles_ 
+instead of updating existing ones. Create and deploy a new version of the existing Pod Template, changing the
+profile name to the newly created profile. Redeploy, once working delete the former Pod Template. This 
+will avoid disruption on in-flight workloads.
+
+The current behavior around updating profiles while they are being used, is not adequate for a built-in 
+security feature such as seccomp. However, given its current "pseudo-GA" state, we don't want to change it 
+with this KEP. We will explore improvements to the behavior through the seccomp-operator and/or a new 
+feature-gated improvement.
+
+###### Profile files managed by the users 
+The current support relies on profiles being saved as files in all cluster nodes where the pods using 
+them may be scheduled. It is also the user's responsibility to ensure the files are correctly saved and 
+synchronised across the all nodes. 
+
+Users can build their own solutions to keep profiles in sync across nodes (e.g. 
+[using daemonsets](https://gardener.cloud/050-tutorials/content/howto/secure-seccomp/)). Or use 
+community driven projects (e.g. [seccomp config](https://github.com/UKHomeOffice/seccomp-config), 
+[openshift's machine config operator](https://github.com/openshift/machine-config-operator), 
+[seccomp operator](https://github.com/saschagrunert/seccomp-operator)) that focuses on solving similar 
+problems.
+
+Promoting LocalhostProfile to GA signals to the community that this is a feature they can use and rely on, 
+until a better option materializes. If and when that happens, a new KEP will be created accordingly.
```

Comment:
Forward-looking statements like this aren't particularly helpful. If solving seccomp profile installation is explicitly a non-goal, put that in the non-goals section, but don't promise future enhancements here.

Reply:
since we are trying to do a kep per feature, any enhancement post GA is new feature and new kep, so agree that we dont need to promise future enhancements.